datatable/extras/ftrl.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+#-------------------------------------------------------------------------------
+# Copyright 2018 H2O.ai
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#-------------------------------------------------------------------------------
+import datatable as dt
+from datatable.lib import core
+
+
+def ftrl(df_train, df_test, a=0.01, b=1.0, l1=0.0, l2=1.0, 
```

Comment:
This function is probably not needed at all.
You can simply say `ftrl = core.ftrl`.
This is probably true for the `aggregate` function above too.

Reply:
I would prefer to keep the both for the moment, because 
- while debugging it is easier for me to change default parameters without re-compiling the whole thing;
- one can do `help(ftrl)` or `help(aggregate)` to get some help, not sure if this could be done with `core`.

Reply:
sure, we can always remove them later